projects/angular/src/layout/nav/aria-current-link.spec.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { Component } from '@angular/core';
+import { ComponentFixture, fakeAsync, TestBed, tick } from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+import { Router } from '@angular/router';
+import { RouterTestingModule } from '@angular/router/testing';
+
+import { MainContainerWillyWonka } from './chocolate/main-container-willy-wonka';
+import { ClrNavigationModule } from './navigation.module';
+import { ClrAriaCurrentLink } from './aria-current-link';
+@Component({
+  template: `
+    <clr-header class="header">
+      <div class="header-nav">
+        <a clrAriaCurrentLink class="nav-link" routerLink="/home" routerLinkActive="active"
+          ><span class="nav-text">Components</span></a
+        >
+        <a clrAriaCurrentLink class="nav-link" routerLink="/patterns" routerLinkActive="active"
+          ><span class="nav-text">Patterns</span></a
+        >
+      </div>
+    </clr-header>
+  `,
+})
+class TestComponent {}
+
+describe('AriaCurrentLink Directive', () => {
+  let fixture: ComponentFixture<any>;
+  let router: Router;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        ClrNavigationModule,
+        RouterTestingModule.withRoutes([
+          { path: 'home', component: TestComponent },
+          { path: 'patterns', component: TestComponent },
+        ]),
+      ],
+      declarations: [TestComponent, ClrAriaCurrentLink],
+      providers: [MainContainerWillyWonka],
+    });
+
+    fixture = TestBed.createComponent(TestComponent);
+    router = TestBed.inject(Router);
+    fixture.detectChanges();
+  });
+
+  afterEach(() => {
+    fixture.destroy();
+  });
+
+  it('decorates the active link with aria-current', fakeAsync(() => {
+    router.navigate(['/home']);
+    tick();
+    fixture.detectChanges();
+
+    let currentPageElements = fixture.debugElement.queryAll(By.css('[aria-current="page"]'));
+    expect(currentPageElements.length).toEqual(1);
+    expect(currentPageElements[0].nativeElement.href).toMatch(/home/);
+
+    router.navigate(['/patterns']);
+    tick();
+    fixture.detectChanges();
```

Comment:
does this test need fakeAsync/tick here in this test?

Reply:
It does. I don't exactly know why, but I think it's to give the router time to update and add the active status.

And then detectChanges is necessary after the active class is set for the dom to be updated after the subscription in the directive executes. It fails if I comment out either one.